java/src/main/native/include/jni_utils.hpp

```diff
@@ -790,19 +760,43 @@ inline void jni_cuda_check(JNIEnv *const env, cudaError_t cuda_status) {
     JNI_THROW_NEW(env, class_name, message, ret_val)                                               \
   }
 
-#define JNI_CUDA_TRY(env, ret_val, call)                                                           \
-  {                                                                                                \
+// Throw a new exception only if one is not pending then always return with the specified value
+#define JNI_CHECK_THROW_NEW_CUDA_ERROR(env, class_name, e, is_halt, ret_val)                       \
```

Comment:
I'm not a fan of the "is_halt" parameter which drastically alters the behavior of this macro from the caller's perspective (i.e.: it either allows continuation or not).  I'd rather see the main body of this macro factored out into a reusable function that creates and posts the Java exception to the JVM and then returns, and then there's two separate macros, one that always throws a C++ exception and one that does not.

Reply:
I divided them into two separate marcos.